collectors/APICalls/TrackerTracker.js

```diff
@@ -149,107 +216,135 @@ class TrackerTracker {
     }
 
     /**
-     * Return top non-anonymous source from the stack trace.
-     * 
-     * @param {string} stack JS stack trace 
+     * Return top non-anonymous source from Runtime.StackTrace.
+     *
+     * @param {import('devtools-protocol/types/protocol').Protocol.Runtime.StackTrace} params
      * @returns {string}
      */
-    _getScriptURL(stack) {
-        if (typeof stack !== "string") {
-            this._log('⚠️ stack is not a string');
-            return null;
-        }
-
-        const lines = stack.split('\n');
-
-        for (let line of lines) {
-            const lineData = line.match(STACK_SOURCE_REGEX);
-
-            if (lineData) {
-                return lineData[1];
+    _getScriptURLFromStackTrace(params) {
+        if (params.callFrames) {
+            for (const frame of params.callFrames) {
+                const fileUrl = frame.scriptId && this._scriptIdToUrl.get(frame.scriptId);
+                const frameUrl = frame.url;
+                for (const u of [frameUrl, fileUrl]) {
+                    if (u && u !== this._mainURL && u.match(HTTP_URL_REGEX)) {
+                        return u;
+                    }
+                }
             }
         }
-
+        if (params.parent) {
+            return this._getScriptURLFromStackTrace(params.parent);
+        }
         return null;
     }
 
     /**
-     * @param {string} breakpointName
-     * @returns {import('./breakpoints').MethodBreakpoint|import('./breakpoints').PropertyBreakpoint} 
+     * Return top non-anonymous source from the Debugger.paused event
+     *
+     * @param {import('devtools-protocol/types/protocol').Protocol.Debugger.PausedEvent} params
+     * @returns {string}
      */
-    _getBreakpointByName(breakpointName) {
-        for (let breakpoint of allBreakpoints) {
-            const {proto, global, props, methods} = breakpoint;
-            const obj = global || `${proto}.prototype`;
-            const matchingProp = props.find(prop => (prop.description || `${obj}.${prop.name}`) === breakpointName);
-
-            if (matchingProp) {
-                return matchingProp;
+    _getScriptURLFromPausedEvent(params) {
+        let script = null;
+        if (params.callFrames) {
+            for (const frame of params.callFrames) {
+                const locationUrl = frame.location && this._scriptIdToUrl.get(frame.location.scriptId);
+                const functionLocationUrl = frame.functionLocation && this._scriptIdToUrl.get(frame.functionLocation.scriptId);
+                const frameUrl = frame.url; // this is usually empty in Debugger.CallFrame (unlike Runtime.CallFrame)
+
+                for (const u of [frameUrl, functionLocationUrl, locationUrl]) {
+                    if (u && u !== this._mainURL && u.match(HTTP_URL_REGEX)) {
+                        script = u;
+                        break;
```

Comment:
break statement can have a label making it a goto function almost - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/break - so you could jump two hops up w/o that additional if below. Not pushing for it, it's a bit obscure, but for your consideration.

Reply:
Hm, I have never used this before, but it's kinda neat, so I changed it